diff --git a/test/function/body_must_be_block.lox b/test/function/body_must_be_block.lox
index 745a8d0..16f0337 100755
--- a/test/function/body_must_be_block.lox
+++ b/test/function/body_must_be_block.lox
@@ -1,3 +1,5 @@
-// [line 3] Error at '123': Expect '{' before block.
-// [c line 4] Error at end: Expect '}' after block.
+// original was -> [line 3] Error at '123': Expect '{' before block.
+// but because we support lambdas the error is slightly different:
+// [line 5] Error at '123': Expect '{' before function body.
+// ?? [c line 4] Error at end: Expect '}' after block.
 fun f() 123;
diff --git a/test/operator/add_bool_string.lox b/test/operator/add_bool_string.lox
index 04739d5..0c8c87a 100755
--- a/test/operator/add_bool_string.lox
+++ b/test/operator/add_bool_string.lox
@@ -1 +1,2 @@
-true + "s"; // expect runtime error: Operands must be two numbers or two strings.
+// No error here, Chapter 7 challenge 2 implemented.
+// true + "s"; expect runtime error: Operands must be two numbers or two strings.
\ No newline at end of file
diff --git a/test/unexpected_character.lox b/test/unexpected_character.lox
index 5e51396..9ca1244 100755
--- a/test/unexpected_character.lox
+++ b/test/unexpected_character.lox
@@ -1,3 +1,3 @@
-// [line 3] Error: Unexpected character.
+// [line 3] Error: Unexpected character. '|'
 // [java line 3] Error at 'b': Expect ')' after arguments.
 foo(a | b);
